config/jupyterhub_config.py

```diff
@@ -4,24 +4,40 @@
 https://jupyterhub.readthedocs.io/en/latest/tutorial/getting-started/config-basics.html
 https://jupyterhub.readthedocs.io/en/stable/reference/config-reference.html
 """
+
 import os
 
+import nativeauthenticator
+
 from jupyterhub_config.custom_spawner import VirtualEnvSpawner
 
 c = get_config()
 
-# Set the authenticator class
-# TODO: Change the authenticator class to a secure one (e.g. GitHubOAuthenticator)
-c.JupyterHub.authenticator_class = 'jupyterhub.auth.DummyAuthenticator'
-c.Authenticator.allowed_users = {'spark_user', 'test_user1', 'test_user2'}
-c.DummyAuthenticator.password = os.environ['JUPYTERHUB_ADMIN_PASSWORD']
+# Set the authenticator class to nativeauthenticator
+# ref: https://native-authenticator.readthedocs.io/en/latest/quickstart.html
+c.JupyterHub.authenticator_class = 'native'
+c.JupyterHub.template_paths = [f"{os.path.dirname(nativeauthenticator.__file__)}/templates/"]
+c.NativeAuthenticator.open_signup = True
+c.NativeAuthenticator.check_common_password = True
+c.NativeAuthenticator.minimum_password_length = 6
```

Comment:
Way too short in general, but for a stopgap I guess it doesn't matter

Reply:
changed to 8.

Reply:
I mean a decent password should be >16 or so characters, 8 is going to be in rainbow tables, but like I say if this is temporary and not exposed to the network is probably doesn't matter.